### ISSUE TITLE
How do I encode OOV ClassLabel in my test set?

### ISSUE THREAD

**Issue by dakl** [tensorflow/datasets#3290]
👋 

I have a dataset of interactions, with a temporal split so that the test set is distinct in time from train and validation sets. This means that I have _unseen_ users in my test set. I've been dabbling with encoding my user_ids and item_ids as ClassLabels, making it easy to get a distinct list of user_ids and item_ids use in embedding layers.

Now, what happens is that the builder raises when rendering the test set, due to the ClassLabels being based on train/val only (which is what I want, for the test set to reflect as true test set as possible, with previously unseen users being in there).

Is there a better way of doing this? Or is there an option that I haven't found to encode out-of-vocabulary labels?

I've been trying both `Text` and `tf.string` as well, but really like ClassLabel since I can easily get lists of unique IDs when loading the dataset.

**Environment information**
(if applicable)
* Operating System: macOs
* Python version: 3.7.7
* `tensorflow-datasets`/`tfds-nightly` version: tensorflow-datasets==4.3.0
* `tensorflow`/`tensorflow-gpu`/`tf-nightly`/`tf-nightly-gpu` version: tensorboard==2.5.0

**Comment by dakl** [comment]
Oh, one more q: is there a hook I can use to write ID lists in the metadata after it has been rendered? I can't find any in the docs, but I might be missing it :D

**Comment by Conchylicultor** [comment]
Besides using string, the only way of doing this is to have `ClassLabel(name=train_ids + test_ids + validation_ids)`, with the test_ids values only used for the test split. ClassLabel is a mapping str <> int, and need to know the full set at the beginning. There is no way to add additional string afterward.

Otherwise, you could set the label values to `-1` in the test set. `-1` means unknown/no-label.

**Comment by dakl** [comment]
Thanks! I'll roll with string and eat having to get unique items after the fact!
